### PR TITLE
d/docdb_engine_version: New data source

### LINKED ISSUE #15255: d/docdb_engine_version: New data source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Ease the practitioner burden of working across regions and partitions by providing access to information on DocDB database engine versions and specifically the default version. In addition, the `preferred_versions` argument should allow selecting a version from a list of acceptable options.

This data source is part of helping practitioners to avoid this error:

```
Error creating DB Instance: InvalidParameterCombination: DocDB does not support creating a DB instance with the
following combination: DBInstanceClass=db.t1.micro, Engine=mysql, EngineVersion=5.6.27,
LicenseModel=general-public-license. For supported combinations of instance class and database engine version,
see the documentation.
```

### New Data Source

* aws_docdb_engine_version

### Potential Terraform Configuration

```hcl
data "aws_docdb_engine_version" "test" {}
```

### References

* hashicorp/terraform#17013
* #15205
* [d/docdb_orderable_db_instance](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/docdb_orderable_db_instance)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #15255
Relates #15228, #15259 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
d/docdb_engine_version: New data source
```

The output from acceptance testing (GovCloud):

```
--- PASS: TestAccAWSDocDBEngineVersionDataSource_basic (15.96s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_preferred (17.53s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_defaultOnly (17.53s)
```

The output from acceptance testing (commercial):

```
--- PASS: TestAccAWSDocDBEngineVersionDataSource_basic (11.66s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_preferred (12.77s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_defaultOnly (12.77s)
```

### CODE REVIEW

**Comment by YakDriver** [comment]
Should we note an AWS bug in the docs? This seems to uncover a bug in the AWS API. If `DBParameterGroupFamily` is set to a valid value, AWS does not return any versions. This holds true for AWS SDK Go and the AWS CLI:

```console
% aws docdb describe-db-engine-versions \
 --engine docdb \
 --query "DBEngineVersions[] [DBParameterGroupFamily]" \
 --output text | sort -u
docdb3.6
% aws docdb describe-db-engine-versions \
 --db-parameter-group-family docdb3.6  

{
    "DBEngineVersions": []
}
```

File: aws/data_source_aws_docdb_engine_version.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			},
+
+			"exportable_log_types": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+				Set:      schema.HashString,
+			},
+
+			"parameter_group_name": {
```
Comment (bflad) [inline aws/data_source_aws_docdb_engine_version.go:34]: Similar note as #15228 here and the rest of this pull request:

```suggestion
			"parameter_group_family": {
```

File: website/docs/d/docdb_engine_version.markdown [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* `version` - (Optional) Version of the DB engine. For example, `3.6.0`. If `version` and `preferred_versions` are not set, the data source will provide information for the AWS-defined default version.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `engine_description` - The description of the database engine.
+* `exportable_log_types` - The types of logs that the database engine has available for export to CloudWatch Logs.
+* `supports_log_exports_to_cloudwatch` - Indicates whether the engine version supports exporting the log types specified by `exportable_log_types` to CloudWatch Logs.
+* `valid_upgrade_targets` - A list of engine versions that this database engine version can be upgraded to.
```
Comment (bflad) [inline website/docs/d/docdb_engine_version.markdown:37]: Similar note as #15228 here:

```suggestion
* `exportable_log_types` - Set of log types that the database engine has available for export to CloudWatch Logs.
* `supports_log_exports_to_cloudwatch` - Indicates whether the engine version supports exporting the log types specified by `exportable_log_types` to CloudWatch Logs.
* `valid_upgrade_targets` - Set of engine versions that this database engine version can be upgraded to.
```

**Review by bflad — APPROVED** [review]
Looks good! Similar comments as #15228 🚀 

Output from acceptance testing:

```
--- PASS: TestAccAWSDocDBEngineVersionDataSource_defaultOnly (14.08s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_basic (14.11s)
--- PASS: TestAccAWSDocDBEngineVersionDataSource_preferred (14.28s)
```

**Pull request merged**